google/cloud/forseti/services/inventory/base/gcp.py

```diff
@@ -1003,22 +1003,28 @@ def _create_storage(self):
         return storage.StorageClient(self.config)
 
     @create_lazy('bigquery', _create_bq)
-    def fetch_bigquery_dataset_policy(self, project_number, dataset_id):
+    def fetch_bigquery_dataset_policy(self, project_id, project_number, dataset_id):
         """Dataset policy Iterator for a dataset from gcp API call.
 
         Args:
+            project_id (str): id of the project to query.
             project_number (str): number of the project to query.
             dataset_id (str): id of the dataset to query.
 
         Returns:
             dict: Dataset Policy.
         """
+        if project_id:
```

Comment:
Since the API can always use the project_number, all you need here is

del project_id  # Live API call uses the project_number
return self.bigquery.get_dataset_access(project_number, dataset_id)

Reply:
Updated.